### PR TITLE
RFC: secret-retrieval with summon and conjur-api-ruby

### PR DESCRIPTION
**Pull request by doodlesbykumbi**
+ uses `summon` and `conjur-api-ruby` for secret retrieval.
+ provides examples of entitlement revocation using policy replacement
+ requires only docker to run

This pattern seems would be good to create a blog post from.
Comments and suggestions welcome!

### CODE REVIEW

File: test_app/Dockerfile [hunk 1]
```diff
@@ -7,15 +7,18 @@ RUN apt-get update -y && \
...
 WORKDIR /opt/app
 
 COPY Gemfile /opt/app/
-COPY Gemfile.lock /opt/app/
 RUN bundle install
 
 COPY . /opt/app
+
+RUN mkdir -p /usr/local/lib/summon
+COPY summon-conjur /usr/local/lib/summon
```
Comment (jvanderhoof) [inline test_app/Dockerfile]: I'd strong prefer to get this moved into our Ansible Role.  It feels like cheating to copy it into the container. :)

File: 2_set_secrets.sh [hunk 2]
```diff
@@ -2,21 +2,14 @@
...
+conjur variable values add staging/myapp/database/username foo_staging_user
+conjur variable values add staging/myapp/database/password "staging_$(openssl rand -hex 12)"
+conjur variable values add staging/myapp/stripe/private_key "staging_$(openssl rand -hex 60)"
+
+conjur variable values add production/myapp/database/username foo_production_user
+conjur variable values add production/myapp/database/password "production_$(openssl rand -hex 12)"
+conjur variable values add production/myapp/stripe/private_key "production_$(openssl rand -hex 60)"
+EOF
+)
+docker-compose run --rm -e CONJUR_AUTHN_API_KEY=$api_key --entrypoint="bash -c '$command'" conjur_cli
```
Comment (jvanderhoof) [inline 2_set_secrets.sh]: Another option to format this is: 
```
docker-compose run --rm -e CONJUR_AUTHN_API_KEY=$api_key --entrypoint /bin/bash conjur_cli -c "
  conjur variable values add staging/myapp/database/username foo_staging_user
  conjur variable values add staging/myapp/database/password "staging_$(openssl rand -hex 12)"
  conjur variable values add staging/myapp/stripe/private_key "staging_$(openssl rand -hex 60)"
  conjur variable values add production/myapp/database/username foo_production_user
  conjur variable values add production/myapp/database/password "production_$(openssl rand -hex 12)"
  conjur variable values add production/myapp/stripe/private_key "production_$(openssl rand -hex 60)
"

File: test_app/Dockerfile (hunk 1 above)
Comment (doodlesbykumbi) [reply test_app/Dockerfile]: @jvanderhoof i've gone ahead and used the `summon` and `summon-conjur` install scripts in the playbook. i think that's a bit more fair. 

here's a thought! users might not trust running a script that calls out to the net to install summon and so might opt to build the binaries themselves and just have a step in the playbook to copy them into the remote hosts.

File: 2_set_secrets.sh (hunk 2 above)
Comment (doodlesbykumbi) [reply 2_set_secrets.sh]: thank you! i've been using `cat` here, `cat` there, `cat` everywhere + all this `EOF` stuff. i'd somehow convinced myself that^ wouldn't work.

File: proxy/ssl.conf [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+x509_extensions = v3_ca # The extentions to add to the self signed cert
+req_extensions  = v3_req
+x509_extensions = usr_cert
+
+[ dn ]
+C=IL
+ST=Israel
+L=TLV
+O=Onyx
+OU=CyberArk
```
Comment (dividedmind) [inline proxy/ssl.conf]: I think these should be the other way around: O is organization, OU is organizational unit. Just saying ;)

Comment (doodlesbykumbi) [reply proxy/ssl.conf]: such attention to detail. thanks

**Pull request closed without merging**